lib/puppet/functions/katello/certificate_subject.rb

```diff
@@ -0,0 +1,23 @@
+# @summary
+#   Extracts a certificate's subject in RFC2253 form
+#
+# @example How to extract a certificate's subject
+#   $cert_subject = katello::certificate_subject($path_to_certificate)
+#
+Puppet::Functions.create_function(:'katello::certificate_subject') do
+  # @param certificate_path
+  #
+  # @return [String]
+  dispatch :certificate_subject do
+    param 'Stdlib::Absolutepath', :certificate_path
```

Comment:
It can return `nil`. It's also better to really use a return type.

And I also made a note that functions used in Deferred can't use custom types but only built in ones.

```suggestion
  dispatch :certificate_subject do
    param 'String[1]', :certificate_path
    return_type 'Optional[String]'
```